spec/models/user_spec.rb

```diff
@@ -136,9 +136,22 @@
       expect(existing_user.renamed_at).to eq(Time.current)
     end
 
-    it "does not set renamed_at when username is not changed" do
-      existing_user.update(email: "newemail@example.com")
-      expect(existing_user.renamed_at).to be_nil
+    context "email is changed" do
```

Comment:
As [recommended by Better Specs](https://www.betterspecs.org/#contexts), we generally start our `context` descriptions with `when` (`with` and `without` are also very good).

It would probably also make sense to move these tests after the ones relating to renames -- right now it's separating `it "sets renamed_at if username is changed"` from all of the other rename-related tests.